wled00/e131.cpp

```diff
@@ -194,35 +194,36 @@ void handleE131Packet(e131_packet_t* p, IPAddress clientIP, byte protocol){
-            effectCurrent = e131_data[dataOffset+ 1];
-          effectSpeed     = e131_data[dataOffset+ 2]; // flickers
-          effectIntensity = e131_data[dataOffset+ 3];
-          effectPalette   = e131_data[dataOffset+ 4];
-          col[0]          = e131_data[dataOffset+ 5];
-          col[1]          = e131_data[dataOffset+ 6];
-          col[2]          = e131_data[dataOffset+ 7];
-          colSec[0]       = e131_data[dataOffset+ 8];
-          colSec[1]       = e131_data[dataOffset+ 9];
-          colSec[2]       = e131_data[dataOffset+10];
+            if (e131_data[dataOffset+1] != seg.mode)  {strip.setMode(id,  e131_data[dataOffset+1]); stateChanged = true;}
+          if (e131_data[dataOffset+2]   != seg.speed)     {seg.speed     = e131_data[dataOffset+2]; stateChanged = true;}      
+          if (e131_data[dataOffset+3]   != seg.intensity) {seg.intensity = e131_data[dataOffset+3]; stateChanged = true;}
+          if (e131_data[dataOffset+4]   != seg.palette)   {seg.palette   = e131_data[dataOffset+4]; stateChanged = true;}
+
+          uint32_t colors[2];
+          byte whites[2] = {0,0};
           if (dmxEffectChannels == 13) {
-            col[3]        = e131_data[dataOffset+11]; // white
-            colSec[3]     = e131_data[dataOffset+12];
+            whites[0] = e131_data[dataOffset+11];
+            whites[1] = e131_data[dataOffset+12];
           }
+          colors[0] = RGBW32(e131_data[dataOffset+5], e131_data[dataOffset+6], e131_data[dataOffset+ 7], whites[0]);
+          colors[1] = RGBW32(e131_data[dataOffset+8], e131_data[dataOffset+9], e131_data[dataOffset+10], whites[1]);
+          if (colors[0] != seg.colors[0]) {seg.setColor(0, colors[0]); stateChanged = true;}
+          if (colors[1] != seg.colors[1]) {seg.setColor(1, colors[1]); stateChanged = true;}
+
+          // Set segment opacity or global brightness
           if (isSegmentMode) {
-            opacity = e131_data[dataOffset+ 0];
-            segmentUpdated |= applyValuesToSegment(seg);
-          } else {
-            if (bri != e131_data[dataOffset+0])
-              bri = e131_data[dataOffset+0];
-            transitionDelayTemp = 0;                  // act fast
-            colorUpdated(CALL_MODE_NOTIFICATION);     // don't send UDP
-            return;                                   // don't activate realtime live mode
+            if (e131_data[dataOffset]   != seg.opacity)   {seg.opacity   = e131_data[dataOffset]; stateChanged = true;}
+          } else if ( id == strip.getSegmentsNum()-1 ) {
+            if (bri != e131_data[dataOffset]) {
+              bri = e131_data[dataOffset];
+              strip.setBrightness(bri, true);
+            }
           }
         }
-        if (segmentUpdated) {
-          transitionDelayTemp = 0;
-          stateUpdated(CALL_MODE_NOTIFICATION);
-          return;
+        if (stateChanged) {
+          transitionDelayTemp = 0;                      // act fast
```

Comment:
Not sure if this will help since transitions are started when changing segments.
It may also not be desired as it is inconsistent with regular behaviour.

Reply:
I'll test if `transitionDelayTemp = 0` has any impact here. It's in there because it was there in past. As you said, this could be obsolete, I'll find out.. Or maybe this needs to be placed before every `seg.setXXX()` method? Nevertheless, to call `stateUpdated` here I think is better than applying single channel changes immediately by `seg.setXXX()` methods (for every segment), triggering transitions. Maybe it's better to explicitly avoid transitions if possible. Starting transitions of DMX segment changes could impact performance of remote control, especially for larger amount of segments.

Reply:
If you do not want transitions use `strip.setTransition(0);`. It will affect all segments, all kinds of transitions.